ipynb/chromeos/ChromeOS_Tests.ipynb

```diff
@@ -0,0 +1,827 @@
+      "2016-04-19 10:15:28,589 INFO    :         Target - Topology:\n",
+      "2016-04-19 10:15:28,590 INFO    :         Target -    [[0, 1], [2, 3]]\n",
+      "2016-04-19 10:15:28,691 DEBUG   : sudo -- sh -c 'cat '\\''/sys/devices/system/cpu/online'\\'''\n",
+      "2016-04-19 10:15:29,597 DEBUG   : cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_available_frequencies\n",
+      "2016-04-19 10:15:30,014 DEBUG   : sudo -- sh -c 'cat '\\''/sys/devices/system/cpu/online'\\'''\n",
+      "2016-04-19 10:15:30,920 DEBUG   : cat /sys/devices/system/cpu/cpu2/cpufreq/scaling_available_frequencies\n",
+      "2016-04-19 10:15:31,237 DEBUG   :       Platform - Trying to load default EM from /home/pippo/work/lisa/libs/utils/platforms/oak.json\n",
+      "2016-04-19 10:15:31,238 DEBUG   :       Platform - Platform descriptor initialized\n",
+      "{'nrg_model': None, 'clusters': {'big': [2, 3], 'little': [0, 1]}, 'cpus_count': 4, 'freqs': {'big': [507000, 702000, 1001000, 1209000, 1404000, 1612000, 1807000, 2106000], 'little': [507000, 702000, 1001000, 1105000, 1209000, 1300000, 1508000, 1703000]}, 'topology': [[0, 1], [2, 3]]}\n",
+      "2016-04-19 10:15:31,254 DEBUG   : /usr/bin/scp -r   /home/pippo/work/lisa/libs/devlib/devlib/bin/arm64/trace-cmd root@192.168.0.1:/root/devlib-target/bin/trace-cmd\n",
+      "2016-04-19 10:15:31,711 DEBUG   : chmod a+x /root/devlib-target/bin/trace-cmd\n",
+      "2016-04-19 10:15:32,132 DEBUG   : cat /sys/kernel/debug/tracing/available_events\n",
+      "2016-04-19 10:15:32,482 INFO    :         FTrace - Enabled tracepoints:\n",
+      "2016-04-19 10:15:32,483 INFO    :         FTrace -   cpu_frequency\n",
+      "2016-04-19 10:15:32,483 INFO    :         FTrace -   cpu_idle\n",
+      "2016-04-19 10:15:32,484 INFO    :         FTrace -   sched_switch\n",
+      "2016-04-19 10:15:32,485 DEBUG   : No RT-App workloads, skipping calibration\n",
+      "2016-04-19 10:15:32,486 INFO    :        TestEnv - Set results folder to:\n",
+      "2016-04-19 10:15:32,486 INFO    :        TestEnv -    /home/pippo/work/lisa/results/20160419_101532\n",
+      "2016-04-19 10:15:32,487 INFO    :        TestEnv - Experiment results available also in:\n",
+      "2016-04-19 10:15:32,488 INFO    :        TestEnv -    /home/pippo/work/lisa/results_latest\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DONE\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Initialize a test environment using:\n",
+    "# the provided target configuration (my_conf)\n",
+    "te = TestEnv(my_conf)\n",
+    "print \"DONE\"\n",
+    "target = te.target"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Get host IP\n",
+    "s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)\n",
+    "s.connect((\"gmail.com\",80))\n",
```

Comment:
Interesting trick! :-)
But we do really need to setup a connection to get the local IP?
This breaks the notebook if we run it without internet access... maybe there is a Python API to get the local IP address?